src/control_node.cpp

```diff
@@ -23,51 +20,49 @@ BT::ControlNode::ControlNode(std::string name) : TreeNode::TreeNode(name)
     // ReturnStatus child_i_status_ = BT::IDLE;  // commented out as unused
 }
 
-void BT::ControlNode::AddChild(TreeNode* child)
+void BT::ControlNode::addChild(TreeNode* child)
 {
-//    // Checking if the child is not already present
-//    for (unsigned int i=0; i<children_nodes_.size(); i++)
-//    {
-//        if (children_nodes_[i] == child)
-//        {
-//            throw BehaviorTreeException("'" + child->Name() + "' is already a '" + get_name() + "' child.");
-//        }
-//    }
+    // Checking if the child is not already present
```

Comment:
Still puzzled whether the check should be there or not. Especially with nodes with memory, someone could find useful to have repeated actions e.g. SequenceMem(Look Object, Approach Object, Look Object, Grasp Object)

Reply:
I want to discuss this in the future but the problem is that we make easier to create circular dependencies if we are not careful. Reusing the same object might have hard to debug consequences, but this PR is not the right time to discuss this.

Your call then: if you want, we can comment this again.

Reply:
I would say let's comment this back and we can discuss that later.